### PR TITLE
LabAssignment3

### PR DESCRIPTION
**Pull request by jantoniosantiz**
deployment and exceptional work complete

### CODE REVIEW

File: LabAssignments/LabAssignment3/LabAssignment3.ipynb [hunk 1]
```diff
@@ -331,13 +356,1801 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Newton's Update Method - Hessian\n",
+    "- In multiple dimensions when dealing with functions of several variables, this involves using the Hessian matrix (a square matrix of second-order partial derivatives of the function) and the gradient of the function. This method is used to find points where the function reaches its local maxima or minima. The inclusion of the Hessian allows Newton's method to consider the curvature of the function, potentially leading to faster convergence compared to methods that only use the first derivative, such as Gradient Descent.\n",
+    "\n",
+    "References:\n",
```
Comment (Owen-Deng) [inline LabAssignments/LabAssignment3/LabAssignment3.ipynb:1379]: here seems a duplicate reference.
@SerenaHangSinclair

File: LabAssignments/LabAssignment3/LabAssignment3.ipynb [hunk 2]
```diff
@@ -331,13 +356,1801 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train and test the model using different feature engineering methods to verify performance improvement\n",
+    "\n",
+    "# Use original data\n",
+    "X = features\n",
```
Comment (Owen-Deng) [inline LabAssignments/LabAssignment3/LabAssignment3.ipynb:368]: The X and y will be overwritten by the following code.

File: LabAssignments/LabAssignment3/LabAssignment3.ipynb [hunk 3]
```diff
@@ -331,13 +356,1801 @@
...
+    "y = labels\n",
+    "\n",
+    "# Use PCA data without cross validation\n",
+    "X = pca_feature\n",
+    "y = labels\n",
+    "\n",
+    "# Use PCA data with cross validation\n",
+    "# binary classification data\n",
+    "for train_indices, test_indices in split_indices:\n",
+    "    X_train, y_train = pca_feature[train_indices], (labels[train_indices] >1)\n",
```
Comment (Owen-Deng) [inline LabAssignments/LabAssignment3/LabAssignment3.ipynb:378]: Similarly, each iteration, the X_train and y_train will be overwritten by next iteration.  
if you are not going to use cross-validation, you can use something like below and remove the loop.
X_train, y_train = pca_feature[split_indices[0][0]], pca_feature[split_indices[0][1]]

File: LabAssignments/LabAssignment3/LabAssignment3.ipynb [hunk 4]
```diff
@@ -331,13 +356,1801 @@
...
+    "    X_test, y_test = pca_feature[test_indices], (labels[test_indices] > 1)\n",
+    "# multi-class classification data\n",
+    "for train_indices, test_indices in split_indices:\n",
+    "    X_train, y_train = pca_feature[train_indices], labels[train_indices]\n",
+    "    X_test, y_test = pca_feature[test_indices], labels[test_indices]\n"
```
Comment (Owen-Deng) [inline LabAssignments/LabAssignment3/LabAssignment3.ipynb:383]: X_train, y_train, X_test, y_test will be overwritten.

**Review by SerenaHangSinclair — COMMENTED** [review]
reviewed

**Pull request merged**